internal/build/error.go

```diff
@@ -56,8 +56,28 @@ func (e RunStepFailure) Error() string {
 }
 
 func IsRunStepFailure(err error) bool {
-	_, ok := err.(RunStepFailure)
+	_, ok := MaybeRunStepFailure(err)
 	return ok
 }
 
+func MaybeRunStepFailure(err error) (RunStepFailure, bool) {
```

Comment:
this seems a little clunky, is there a more elegant way to do this? (Alternately, it seems maybe overkill to recursively drill into causes rather than just checking the very first cause and leaving it at that... but also, ideally no matter how many times we wrap an error, it should still be detectable as a rsf, sooo 🤷‍♀ )

Reply:
fyi, the Go2 errors proposal has a library for this
https://godoc.org/golang.org/x/exp/errors#Is